packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/project/services/ProjectSemanticDataExportParticipant.java

```diff
@@ -0,0 +1,125 @@
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.application.project.services;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import org.eclipse.emf.ecore.EPackage;
+import org.eclipse.sirius.components.core.api.IEditingContextSearchService;
+import org.eclipse.sirius.components.emf.ResourceMetadataAdapter;
+import org.eclipse.sirius.components.emf.services.api.IEMFEditingContext;
+import org.eclipse.sirius.emfjson.resource.JsonResourceFactoryImpl;
+import org.eclipse.sirius.web.application.UUIDParser;
+import org.eclipse.sirius.web.application.document.services.api.IDocumentExporter;
+import org.eclipse.sirius.web.application.project.services.api.IProjectExportParticipant;
+import org.eclipse.sirius.web.domain.boundedcontexts.project.Project;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+
+/**
+ * Used to add semantic data to the export of a project.
+ *
+ * @author sbegaudeau
+ */
+@Service
+public class ProjectSemanticDataExportParticipant implements IProjectExportParticipant {
+
+    private final IEditingContextSearchService editingContextSearchService;
+
+    private final List<IDocumentExporter> documentExporters;
+
+    private final Logger logger = LoggerFactory.getLogger(ProjectSemanticDataExportParticipant.class);
+
+    public ProjectSemanticDataExportParticipant(IEditingContextSearchService editingContextSearchService, List<IDocumentExporter> documentExporters) {
+        this.editingContextSearchService = Objects.requireNonNull(editingContextSearchService);
+        this.documentExporters = Objects.requireNonNull(documentExporters);
+    }
+
+    @Override
+    public Map<String, Object> exportData(Project project, ZipOutputStream outputStream) {
+        Map<String, Object> manifestEntries = new HashMap<>();
+
+        var optionalEditingContext = this.editingContextSearchService.findById(project.getId().toString())
+                .filter(IEMFEditingContext.class::isInstance)
+                .map(IEMFEditingContext.class::cast);
+        if (optionalEditingContext.isPresent()) {
+            var editingContext = optionalEditingContext.get();
+
+            List<String> metamodels = this.getMetamodels(editingContext);
+            Map<String, String> id2DocumentName = this.exportSemanticData(editingContext, project.getName(), outputStream);
+
+            manifestEntries.put("metamodels", metamodels);
+            manifestEntries.put("documentIdsToName", id2DocumentName);
+        }
+
+        return manifestEntries;
+    }
+
+    private List<String> getMetamodels(IEMFEditingContext editingContext) {
```

Comment:
This is the list of metamodels visible/accesible to the editing context, but not necessarily the ones actually used in it (which can be just a subset). For example for a studio which simply defines a Diagram I get all of:
```
[
  "http://www.eclipse.org/sirius-web/domain",
  "http://www.eclipse.org/sirius-web/deck",
  "http://www.eclipse.org/sirius-web/diagram",
  "http://www.eclipse.org/sirius-web/view",
  "http://www.eclipse.org/sirius-web/gantt",
  "https://www.eclipse.org/sirius/widgets/reference",
  "http://www.eclipse.org/sirius-web/form",
  "http://www.eclipse.org/sirius-web/customnodes"
]
```